api/BUILD.bazel

```diff
@@ -0,0 +1,14 @@
+java_library(
+    name = "api",
+    srcs = glob([
+        "src/main/java/**/*.java",
+    ]),
+    visibility = ["//visibility:public"],
+    deps = [
+        "//context",
+        "@com_google_code_findbugs_jsr305//jar",
+        "@com_google_guava_failureaccess//jar",  # future transitive dep of Guava. See #5214
+        "@com_google_guava_guava//jar",
+        "@com_google_j2objc_j2objc_annotations//jar",
```

Comment:
this doesn't obviously match the build.gradle. is that intentional? I can see these deps are mostly the same as what context brings though.. 

```
    deps = [
        "@com_google_code_findbugs_jsr305//jar",
        "@com_google_errorprone_error_prone_annotations//jar",
        "@com_google_guava_guava//jar",
        "@com_google_j2objc_j2objc_annotations//jar",
],
```

Reply:
Uhhh... grpc-context should have no deps. Those others should be removed. They aren't there in context/build.gradle, so at least that's right. It looks like it was added with too many deps when it was initially created in 6d9e149dcb. Created #5591 to clean it up.

The only non-annotation dependencies of grpc-api are grpc-context and Guava.

The dependencies between gradle and Bazel seem to match here. What may be confusing you is animalsniffer and errorprone. Those are included in build.gradle because we want to use our versions instead of the versions used by Guava (since our versions are newer, and we use those dependencies elsewhere in grpc). So you will see errorprone listed, but then there is the exclude in the `compile (libraries.guava)` section. This is Maven dependencyConvergence goo (although we technically don't support dependencyConvergence, but we still want to support requireUpperBoundDeps and don't have a good tool yet to do so).

Transitive vs direct dependencies are messed up still in Bazel, so that adds confusion (it just confused me). j2objc_annotations, for example, I think is only needed by guava. But the Maven Central glue for Bazel hasn't historically done transitive dependencies well (#5359).